Compile Error on Mac

Throwing exceptions inside C++ destructors is not allowed by Clang. However, the following destructor is still using `PADDLE_ENFORCE` inside themselves, which results in a Mac compile error.

https://github.com/PaddlePaddle/Paddle/blob/4a0761781c10d2409f136b541375cbecbab651a6/paddle/fluid/inference/api/api_impl.cc#L103-L108

```
/Users/baidu/Documents/git_workspace/Paddle/paddle/fluid/inference/api/api_impl.cc:105:5: error: '~NativePaddlePredictor' has a non-throwing exception specification but can still throw [-Werror,-Wexceptions]
    PADDLE_ENFORCE_NOT_NULL(scope_, "Should have parent scope!");
    ^
```

--- a/paddle/fluid/inference/api/api_impl.cc
+++ b/paddle/fluid/inference/api/api_impl.cc
@@ -66,6 +66,7 @@ bool NativePaddlePredictor::Init(
   if (parent_scope) {
     scope_ = parent_scope;
     sub_scope_ = &(parent_scope->NewScope());
+    PADDLE_ENFORCE_NOT_NULL(sub_scope_, "create sub scope fail");
   } else {
     paddle::framework::InitDevices(false);
     scope_.reset(new paddle::framework::Scope());
@@ -102,7 +103,6 @@ bool NativePaddlePredictor::Init(
 
 NativePaddlePredictor::~NativePaddlePredictor() {
   if (sub_scope_) {
-    PADDLE_ENFORCE_NOT_NULL(scope_, "Should have parent scope!");
     scope_->DeleteScope(sub_scope_);
   }
 }

